maxHeap/src/maxHeap.java

```diff
@@ -0,0 +1,84 @@
+public class maxHeap{
+
+    private int heap[];
+
+    private int heapSize;
+
+    public maxHeap(int size){
+
+        heapSize = 0;
+
+        heap = new int [ size + 1 ];
+
+        for ( int i = 0; i < size; i++)
+            heap[i] = -1;
+
+    }
+
+    private int parent( int pos ){
+
+        return (pos-1)/2;
+
+    }
+
+    private void heapifyUp(int pos){
+
+        int temp = heap[pos];
+
+        while( pos > 0 && temp > heap[ parent(pos) ] ){
+
+            heap[pos] = heap[parent(pos)];
+
+            pos = parent(pos);
+
+        }
+
+        heap[pos] = temp;
+
+    }
+
+    public void insert(int n){
+
+        if( heapSize == heap.length )
+            return;
+
+        heap[heapSize++] = n;
+
+        heapifyUp(heapSize-1);
+
+    }
+
+    public void printHeap(){
+
+        System.out.println("Heap is");
+
+        for( int i = 0; i < heapSize; i++ ){
+            System.out.print( heap[i] + " ");
+        }
+
+    }
+
+    public static void main(String[] args) {
+
+        maxHeap dsa = new maxHeap(10);
+
+        dsa.insert(1);
```

Comment:
could have used a loop running 8 times instead, that reduces lines of code.

Reply:
Entries could be some random values as well, so didn't use the loop. But will keep it mind.